src/writers/writer.cpp

```diff
@@ -145,21 +118,65 @@ zarr::FileCreator::create_channel_dirs_(int n_c) noexcept
                            "Failed to create directory: %s",
                            exc.what());
                   err = buf;
+                  failure = true;
```

Comment:
If there is one failure, should we short-circuit other waiting jobs? I imagine failures are often correlated, and we might not want to wait for all of them to fail. Could just check to see if failure is set right before the expensive part, for example.